pykafka/rdkafka/_rd_kafkamodule.c

```diff
@@ -0,0 +1,448 @@
+            "Cannot configure: seems instance was started already?");
+        return NULL;
+    }
+
+    if (! self->rdk_conf) self->rdk_conf = rd_kafka_conf_new();
+    if (! self->rdk_topic_conf) {
+        self->rdk_topic_conf = rd_kafka_topic_conf_new();
+    }
+
+    PyObject *conf_or_topic_conf = topic_conf ? topic_conf : conf;
+    Py_ssize_t i, len = PyList_Size(conf_or_topic_conf);
+    for (i = 0; i != len; ++i) {
+        PyObject *conf_pair = PyList_GetItem(conf_or_topic_conf, i);
+        const char *name = NULL;
+        const char *value =  NULL;
+        if (! PyArg_ParseTuple(conf_pair, "ss", &name, &value)) return NULL;
+
+        char errstr[512];
+        rd_kafka_conf_res_t res;
+        if (topic_conf) {
+            res = rd_kafka_topic_conf_set(
+                    self->rdk_topic_conf, name, value, errstr, sizeof(errstr));
+        } else {
+            res = rd_kafka_conf_set(
+                    self->rdk_conf, name, value, errstr, sizeof(errstr));
+        }
+        if (res != RD_KAFKA_CONF_OK) {
+            set_PyRdKafkaError(RD_KAFKA_RESP_ERR__FAIL, errstr);
+            return NULL;
+        }
+    }
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+
+static PyObject *
+Consumer_start(Consumer *self, PyObject *args, PyObject *kwds) {
+    char *keywords[] = {
+        "brokers",
+        "topic_name",
+        "partition_ids",
+        "start_offsets",  // same order as partition_ids
+        NULL};
+    const char *brokers = NULL;
+    const char *topic_name = NULL;
+    PyObject *partition_ids = NULL;
+    PyObject *start_offsets= NULL;
+    if (! PyArg_ParseTupleAndKeywords(args,
+                                      kwds,
+                                      "ssOO",
+                                      keywords,
+                                      &brokers,
+                                      &topic_name,
+                                      &partition_ids,
+                                      &start_offsets)) {
+        return NULL;
+    }
+
+    // We'll keep our own copy of partition_ids, because the one handed to us
+    // might be mutable, and weird things could happen if the list used on init
+    // is different than that on dealloc
+    if (self->partition_ids) {
+        set_PyRdKafkaError(RD_KAFKA_RESP_ERR__FAIL, "Already started.");
+        return NULL;
+    }
+    self->partition_ids = PySequence_List(partition_ids);
+    if (! self->partition_ids) return NULL;
+
+    // Configure and start a new RD_KAFKA_CONSUMER
+    char errstr[512];
+    self->rdk_handle = rd_kafka_new(
+            RD_KAFKA_CONSUMER, self->rdk_conf, errstr, sizeof(errstr));
+    self->rdk_conf = NULL;  // deallocated by rd_kafka_new()
+    if (! self->rdk_handle) {
+        set_PyRdKafkaError(RD_KAFKA_RESP_ERR__FAIL, errstr);
+        return NULL;
+    }
+    if (rd_kafka_brokers_add(self->rdk_handle, brokers) == 0) {
+        // XXX add brokers via conf setting instead?
```

Comment:
Ah yes - I couldn't make up my mind here. The only upside of adding them via the config list is that I could throw this if-block away, so less code, but on the other hand - this throws a more descriptive exception, so now that it already works anyway, I think you're right. I'll toss out the comment.